### PR TITLE
(FACT-2656) Add solaris networking facts

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

File: lib/resolvers/solaris/ffi/ffi.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'ffi'
+require_relative 'structs.rb'
+require_relative 'functions.rb'
+
+module Facter
+  module Resolvers
+    module Solaris
```
Comment (Filipovici-Andrei) [inline lib/resolvers/solaris/ffi/ffi.rb:9]: namespace doesn't match the folder structure. This is valid for all files

File: lib/resolvers/solaris/networking_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
+require_relative 'ffi/ffi.rb'
+require 'ipaddr'
+require 'pry'
```
Comment (Filipovici-Andrei) [inline lib/resolvers/solaris/networking_resolver.rb:3]: remove pry

File: lib/resolvers/solaris/networking_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                mtu: load_mtu(socket, lifreq)
+              }
+              Socket::close(socket)
+
+            end
+            @fact_list[:interfaces] = interfaces
+            @fact_list[fact_name]
+          end
+
+          def create_socket(family)
```
Comment (Filipovici-Andrei) [inline lib/resolvers/solaris/networking_resolver.rb:45]: probably something more suggestive than family, would be useful

File: README.md [hunk 4]
```diff
@@ -1,50 +1,2 @@
-# facter-ng
```
Comment (IrimieBogdan) [inline README.md:1]: The original README should be left in place

File: lib/facter/facts/solaris/networking/network6.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Facts
+  module Solaris
+    module Networking
+      class Network6
+        FACT_NAME = 'networking.network'
```
Comment (oanatmaria) [inline lib/facter/facts/solaris/networking/network6.rb:7]: networking.network6, same for alias

File: lib/facter/facts/solaris/networking/network6.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        ALIASES = 'network'
+
```
Comment (oanatmaria) [inline lib/facter/facts/solaris/networking/network6.rb:9]: Could you please add some tests for these facts?

File: lib/facter/facts/solaris/networking/network6.rb (hunk 5 above)
Comment (IrimieBogdan) [reply lib/facter/facts/solaris/networking/network6.rb]: Changed the name and alias.

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris10-64a,ubuntu1604-32a,ubuntu1804-64a,windows2016-64a,windows2019-64a

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris11-64a,ubuntu1604-32a,ubuntu1804-64a,windows2016-64a,windows2019-64a

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian9-64a,debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,redhat7-POWERa,sles12-64a,sles15-64a,solaris11-64a,ubuntu1604-32a,ubuntu1804-64a,windows2016-64a,windows2019-64a

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,sles12-64a,sles15-64a,solaris11-64a,solaris114-64a,ubuntu1604-32a,ubuntu1804-64a

**Comment by mihaibuzgau** [comment]
jenkins please test this on debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,sles12-64a,sles15-64a,solaris11-64a,solaris114-64a,ubuntu1604-32a,ubuntu1804-64a

File: spec/facter/facts/solaris/ipaddress_interfaces_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                        an_object_having_attributes(name: 'ipaddress_en1',
+                                                    value: interfaces['en1'][:ip], type: :legacy))
+    end
+  end
+
+  describe '#call_the_resolver when resolver returns nil' do
+    let(:interfaces) { nil }
+
+    it 'returns nil' do
+      expect(fact.call_the_resolver).to be_an_instance_of(Array).and contain_exactly
```
Comment (Filipovici-Andrei) [inline spec/facter/facts/solaris/ipaddress_interfaces_spec.rb:31]: I think you should check for nil not array

File: spec/facter/facts/solaris/macaddress_interfaces_spec.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                        an_object_having_attributes(name: 'macaddress_en1',
+                                                    value: interfaces['en1'][:mac], type: :legacy))
+    end
+  end
+
+  describe '#call_the_resolver when resolver returns nil' do
+    let(:interfaces) { nil }
+
+    it 'returns nil' do
+      expect(fact.call_the_resolver).to be_an_instance_of(Array).and contain_exactly
```
Comment (Filipovici-Andrei) [inline spec/facter/facts/solaris/macaddress_interfaces_spec.rb:31]: This is valid for multiple other tests that check for nil

**Comment by mihaibuzgau** [comment]
jenkins please test this on debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,sles12-64a,sles15-64a,solaris11-64a,solaris114-64a,ubuntu1604-32a,ubuntu1804-64a

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian10-64a,fedora30-64a,osx1014-64a,redhat5-64a,centos8-64a,redhat7-64a,sles12-64a,sles15-64a,solaris11-64a,solaris114-64a,ubuntu1604-32a,ubuntu1804-64a

**Comment by sebastian-miclea** [comment]
jenkins please test this on debian10-64a,fedora30-64a,osx1014-64a,centos8-64a,redhat7-64a,sles12-64a,sles15-64a,solaris11-64a,solaris114-64a,solaris11-SPARCa,ubuntu1604-32a,ubuntu1804-64a

**Pull request merged**